lib/algorithms/helpers.js

```diff
@@ -97,17 +97,14 @@ exports.setupFallback = function(nodejs, webcrypto, fallback) {
       var args = arguments,
           promise;
 
-      function check(err) {
-        if (0 === err.message.indexOf("unsupported algorithm:")) {
-          impl = fallback;
-          return impl.apply(null, args);
-        }
-
-        return Promise.reject(err);
+      function check(_err) {
+        impl = fallback;
+        return impl.apply(null, args);
       }
 
       try {
         promise = Promise.resolve(nodejs.apply(null, args));
+        promise = promise.catch(check);
```

Comment:
`nodejs.apply(null, args)` is a synchronous operation so i think this statement is unreachable.

Reply:
@aaccenture good point! `nodejs.apply` is a synchronous function that can `throw`. However, it also [returns a promise](https://github.com/cisco/node-jose/blob/37db9e5371dc9a0557767a6a0190e2b0ab5cf3b6/lib/algorithms/ecdh.js#L164) that can `reject`. Hence we need to both wrap the call in a try/catch, and call `promise.catch` here. I'll add a comment to clarify.

Reply:
Thanks for the link, I saw `return Promise.reject()` so this catch here save us from unhandled promise rejection. Legitimate.